apps/explorer/src/components/TransactionBlocksForAddress/TransactionBlocksForAddress.tsx

```diff
@@ -168,9 +169,11 @@ function TransactionBlocksForAddress({
 
                                 filterValue,
                             });
-                            // setCurrentPage(currentPage + 1);
                         }}
-                        hasNext={Boolean(hasNextPage)}
+                        hasNext={
+                            Boolean(hasNextPage) &&
+                            Boolean(data?.pages[currentPage])
```

Comment:
prevents repeated next page fetching + when data hasn't fully loaded

was resulting in page count outpacing loaded page numbers

Reply:
I added a `useCursorPagination` hook that does some of this logic if you want to use that here.